source/write/replace.txt

```diff
@@ -0,0 +1,202 @@
+.. _kotlin-sync-write-replace:
+
+=================
+Replace Documents
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: modify, change, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to perform a **replace
+operation** on a document in a MongoDB collection. A replace operation
+removes all fields and values from a specified document except the
+``_id`` field, and adds new fields and values that you specify. This
+operations differs from an update operation, which changes only
+specified fields in one or more documents.
+
+.. tip:: Learn About Update Operations
+
+   To learn more about update operations, see the
+   :ref:`kotlin-sync-write-update` guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The documents in this collection are modeled by the following {+language+} data class:
+
+.. literalinclude:: /includes/write/replace.kt
+   :start-after: start-data-class
+   :end-before: end-data-class
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB by using the
+``replaceOne()`` method. This method removes all fields except the
+``_id`` field from the first document that matches the query filter. It
+then adds the fields and values you specify to the empty document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+You must pass the following parameters to the ``replaceOne()`` method:
+
+- **Query filter**, which matches which documents to update. To learn
+  more about query filters, see the :ref:`kotlin-sync-specify-query`
+  guide.
+
+- **Replacement document**, which specifies the fields and values that
+  you want to replace the existing fields and values with.
+
+Replace One Document
+--------------------
+
+The following example uses the ``replaceOne()`` method to replace the
+fields and values of a document in which the value of the ``name`` field
+value is ``"Primola Restaurant"``:
+
+.. literalinclude:: /includes/write/replace.kt
+   :start-after: start-replace-one
+   :end-before: end-replace-one
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. important::
+
+   The values of ``_id`` fields are immutable. If your replacement document specifies
+   a value for the ``_id`` field, it must be the same as the ``_id`` value of the
+   existing document.
```

Comment:
Q: If the id value is not the same as the existing document, what happens? An error?